### PR TITLE
gh-109373: Store metadata required for pystats comparison in the json

### PR DESCRIPTION
**Pull request by mdboom**
* Issue: gh-109373

### CODE REVIEW

File: Tools/scripts/summarize_stats.py [hunk 1]
```diff
@@ -28,6 +27,15 @@ def format_ratio(num, den):
...
 
+def ratio_to_float(s):
+    """
+    The inverse of format_ratio.
+    """
+    if s == "":
+        return 0.0
+    else:
+        return float(s[:-1])
+
```
Comment (brandtbucher) [inline Tools/scripts/summarize_stats.py]: This confused me, because it's not the inverse, right? It just turns `"4.2%"` into `4.2` (which *feels* wrong to me since I'd expect `.042`... I know that this is what the old code does, and it's only used as a sort key, so I guess either is fine).

So I'd suggest changing the name and docstring, but I honestly can't think of a good name for this operation...

At least, if this is gonna be its own function, maybe `assert s.endswith("%")` so it's clear what this function is doing?

File: Tools/scripts/summarize_stats.py [hunk 2]
```diff
@@ -179,6 +192,16 @@ def gather_stats(input):
                     value = int(value)
                     stats[key] += value
             stats['__nfiles__'] += 1
+
+        import opcode
```
Comment (brandtbucher) [inline Tools/scripts/summarize_stats.py:201]: Just curious, why was this import moved here? Just to reduce the temptation to use it elsewhere in the file?

**Review by brandtbucher — APPROVED** [review]
Looks good, just one suggestion to maybe improve the funny `ratio_to_float` function. It's minor, though, so I'll merge tomorrow regardless.

File: Tools/scripts/summarize_stats.py (hunk 2 above)
Comment (mdboom) [reply Tools/scripts/summarize_stats.py:201]: Yes, exactly.  Any use of `opcode`, `dis` or other sorts of bytecode introspection things on the "compare two results" path could lead to a subtle, silent error.  I'll add a comment.

File: Tools/scripts/summarize_stats.py (hunk 1 above)
Comment (mdboom) [reply Tools/scripts/summarize_stats.py]: Good point.  I renamed the function and clarified in the docstring.
